improver/categorical/decision_tree.py

```diff
@@ -263,22 +273,16 @@ def prepare_input_cubes(
                         missing_data.append([diagnostic, threshold, condition])
                     else:
                         used_cubes.extend(matched_threshold)
-            else:
-                for diagnostic in diagnostics:
-                    test_condition = iris.Constraint(name=diagnostic)
-                    matched_cube = cubes.extract(test_condition)
-                    if not matched_cube:
-                        if "if_diagnostic_missing" in query:
-                            optional_node_data_missing.append(key)
-                        else:
-                            missing_data.append([diagnostic, threshold, condition])
-                        continue
-                    used_cubes.extend(matched_cube)
 
         if missing_data:
             msg = "Decision Tree input cubes are missing the following required input fields:\n"
-            dyn_msg = "name: {}, threshold: {}, " "spp__relative_to_threshold: {}\n"
             for item in missing_data:
+                if len(item) == 1:
+                    dyn_msg = "name:{}"
+                else:
+                    dyn_msg = (
+                        "name: {}, threshold: {}, " "spp__relative_to_threshold: {}\n"
+                    )
```

Comment:
This feels like an inelegant solution, and probably the wrong solution. The issue is that `item` can be different depending on whether we're using a threshold or not. Ideally, `item` would be of a type that always had all possible sub-items, with any irrelevant sub-items set to None (e.g. a namedtuple or a dataclass, see https://stackoverflow.com/questions/11351032/named-tuple-and-default-values-for-optional-keyword-arguments). At this point in the code, you then work through each possible sub-item and include it in the message string if it is relevant.
If you made the new thing a full class, you could include the string formatting element within it as a `__str__` method.

My suggestion is verging on excessive now. Perhaps it would be best to view this as a learning opportunity rather than a PR response. Take a few minutes to understand my suggestions, then implement or ignore as you wish.

Reply:
I've gone with a different way to fix this. I've moved the generation of the strings for missing data into the main body of the code so that different strings are defined for deterministic and probabilistic nodes.